examples/deepcopy-gen/generators/deepcopy.go

```diff
@@ -578,93 +618,96 @@ func (g *genDeepCopy) doBuiltin(t *types.Type, sw *generator.SnippetWriter) {
 }
 
 func (g *genDeepCopy) doMap(t *types.Type, sw *generator.SnippetWriter) {
+	ut := underlyingType(t)
+	uet := underlyingType(ut.Elem)
+
 	sw.Do("*out = make($.|raw$, len(*in))\n", t)
-	if t.Key.IsAssignable() {
+	if ut.Key.IsAssignable() {
 		switch {
-		case hasDeepCopyMethod(t.Elem):
+		case hasDeepCopyMethod(ut.Elem):
 			sw.Do("for key, val := range *in {\n", nil)
 			sw.Do("(*out)[key] = val.DeepCopy()\n", nil)
 			sw.Do("}\n", nil)
-		case t.Elem.IsAnonymousStruct():
+		case ut.Elem.IsAnonymousStruct(): // not uet here because it needs type cast
 			sw.Do("for key := range *in {\n", nil)
 			sw.Do("(*out)[key] = struct{}{}\n", nil)
 			sw.Do("}\n", nil)
-		case t.Elem.IsAssignable():
+		case uet.IsAssignable():
 			sw.Do("for key, val := range *in {\n", nil)
 			sw.Do("(*out)[key] = val\n", nil)
 			sw.Do("}\n", nil)
-		case t.Elem.Kind == types.Interface:
+		case ut.Elem.Kind == types.Interface: // only call this for non-aliased interfaces
```

Comment:
If `ut.Elem` is an interface then `underlyingType` won't change it. Why not using `uet.Kind` here in that case?

Reply:
Actually this does not matter. The golang parser renders interface aliases as interfaces, without access to the underlying type. I fixed this, added comments and examples.